hkmc2/shared/src/main/scala/hkmc2/codegen/js/JSBuilder.scala

```diff
@@ -455,8 +457,13 @@ class JSBuilder(using TL, State, Ctx) extends CodeBuilder:
     blockPreamble(p.main) -> (imps.mkDocument(doc" # ") :/: returningTerm(p.main, endSemi = false).stripBreaks)
   
   def blockPreamble(t: Block)(using Raise, Scope): Document =
+    
     // TODO document: mutable var assnts require the lookup
     val vars = t.definedVars.toSeq.filter(scope.lookup(_).isEmpty).sortBy(_.uid).iterator.map(l =>
+    
+    // TODO: now should be possible to just write: (currently causes problems to handlers)
+    // val vars = t.definedVars.toSeq.sortBy(_.uid).iterator.map(l =>
```

Comment:
@CAG2Mark @AnsonYeung There are probably many instances where the effect handler transformation reassigns non-mutable symbols instead of simply assigning them. In principle it should be possible to make this change, but it breaks code that uses handlers.

Reply:
Indeed the continuation class implicitly captures the variables so all assignments within continuation should be reassignments really (including the this.pc assignments), even though they might not be assigned or initialized in the first place yet.

A brief remark of why capturing is needed: any function could define other nested function that capture the same variable, and it is important for the continuation class to capture the same one.

Reply:
> Indeed the continuation class implicitly captures the variables so all assignments within continuation should be reassignments really (including the this.pc assignments), even though they might not be assigned or initialized in the first place yet.

I'm not following. Doesn't the continuation class just duplicate the function's implementation? If the function was assigning the variable for the first time (maybe the variable is even immutable), then the continuation class should also be assigning that variable for the first time in the corresponding part of its duplicated code. No?

Reply:
Oh right, so we need proper binding site probably. With the changes in the comment above it would make the initial assignment to create a local variable at the function declaration which is incorrect as the resume function can return a effectsig and resume later which uses the same variable again.

Reply:
Ah right, I see. That's probably why it doesn't work currently.

We should list the variables that are bound within a given MIR block body.